x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/InstanceType.java

```diff
@@ -45,16 +45,47 @@ public static InstanceType fromCostSource(Map<String, Object> source) {
      * @return the {@link InstanceType}
      */
     public static InstanceType fromHostSource(Map<String, Object> source) {
```

Comment:
Ideally we'd not have provider-specific fields in `profiling-hosts` but rather use the [Cloud-specific ECS fields](https://www.elastic.co/guide/en/ecs/current/ecs-cloud.html). That's out of scope for this PR but IMHO the right long-term direction.

Reply:
:+1: We just talked about this and agreed to wrap this into an issue.